src/iteration_function.jl

```diff
@@ -0,0 +1,41 @@
+# nlsolve optimizes subtypes of AbstractDifferentiableMultivariateFunction
+mutable struct IterationFunction{I<:AbstractDDEIntegrator} <: AbstractDifferentiableMultivariateFunction
```

Comment:
"Regular" functions are always encapsulated `DifferentiableMultivariateFunction` by NLsolve. I'm not sure whether there is an advantage in defining a subtype of `AbstractDifferentiableMultivariateFunction` with the self-referential field `f!` instead of just subtyping `Function`